projects/components/src/common/activity-reporter/activity-reporter.spec.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { ActivityReporter, ActivityStatus, ActivityResponse } from './activity-reporter';
+
+let resolve;
+let reject;
+let promise;
+
+let reporter: ActivityReporter;
+
+let start;
+let reportError;
+let reportSuccess;
+
+describe('ActivityReporter', () => {
+    beforeEach(() => {
+        promise = new Promise((promiseResolve, promiseReject) => {
+            resolve = stuff => promiseResolve(stuff);
+            reject = stuff => promiseReject(stuff);
+        });
+
+        reporter = new SimpleReporter();
+        start = spyOn(reporter, 'startActivity');
+        reportError = spyOn(reporter, 'reportError');
+        reportSuccess = spyOn(reporter, 'reportSuccess');
+    });
+
+    class SimpleReporter extends ActivityReporter {
+        reportError(errorText: string): void {}
+
+        reportSuccess(successMessage?: string): void {}
+
+        startActivity(): void {}
+    }
+
+    describe('monitorActivity', () => {
+        it('can monitor an activity with success', done => {
```

Comment:
Please improve the text, when we see the same text in multiple its, it's a sign. "can monitor an activity"

I would expect test cases that say

it displays a default message while the promise is pending
it displays a custom message while the promise is pending
it removes the activity reporter from the screen after the promise resolves
it displays an error on the screen if the promise is rejected

And maybe the same thing for when there are multiple promises (`monitorActivities `).

We may need to chat about this when I come back

Reply:
Reworked the messages a little in both this spec file and the spec file for the two concrete ActivityReporters. Let me know what you think